checkov/terraform/tf_parser.py

```diff
@@ -605,38 +604,48 @@ def handle_variables(
                 hcl_tfvars = file
             elif file.name.endswith(".auto.tfvars.json") or file.name.endswith(".auto.tfvars"):
                 auto_vars_files.append(file)
-            elif vars_files and file.path in vars_files:
-                explicit_var_files.append(file)
             elif file.name.endswith(".tf") or file.name.endswith('.hcl'):  # TODO: add support for .tf.json
                 tf_files_to_load.append(file)
 
         for key, value in self.env_vars.items():
             if not key.startswith("TF_VAR_"):
                 continue
-            self.external_variables_data.append((key[7:], value, f"env:{key}"))
+            self.external_variables_data.append((key[7:], value, f"env:{key}", True))
         if hcl_tfvars:  # terraform.tfvars
             data = load_or_die_quietly(hcl_tfvars, self.out_parsing_errors, clean_definitions=False)
             if data:
-                self.external_variables_data.extend([(k, safe_index(v, 0), hcl_tfvars.path) for k, v in data.items()])
+                self.external_variables_data.extend([(k, safe_index(v, 0), hcl_tfvars.path, False) for k, v in data.items()])
         if json_tfvars:  # terraform.tfvars.json
             data = load_or_die_quietly(json_tfvars, self.out_parsing_errors)
             if data:
-                self.external_variables_data.extend([(k, v, json_tfvars.path) for k, v in data.items()])
+                self.external_variables_data.extend([(k, v, json_tfvars.path, False) for k, v in data.items()])
 
         auto_var_files_to_data = self._load_files(auto_vars_files)
         for var_file, data in sorted(auto_var_files_to_data, key=lambda x: x[0]):
             if data:
-                self.external_variables_data.extend([(k, v, var_file) for k, v in data.items()])
+                self.external_variables_data.extend([(k, v, var_file, False) for k, v in data.items()])
 
-        explicit_var_files_to_data = self._load_files(explicit_var_files)
-        # it's possible that os.scandir returned the var files in a different order than they were specified
+        # Load vars files specified via the '--var-file' option, first
+        # collecting their absolute paths.
         if vars_files:
-            for var_file, data in sorted(explicit_var_files_to_data, key=lambda x: vars_files.index(x[0])):
+            for f in [Path(x).resolve() for x in vars_files]:
```

Comment:
it is wasteful to create a list and then iterate over it, just iterate directly over it. Also using `Path` and then multiple times creating a string out of it is a bad idea, you can use `os.path.abspath()` instead.